### PR TITLE
Tickets/DM-6297 Wrap afw::detection

### PR DESCRIPTION
**Pull request by fred3m**

### CODE REVIEW

File: python/lsst/afw/detection/footprint.cc [hunk 1]
```diff
@@ -22,16 +22,113 @@
...
+    clsFootprint.def("normalize", &Footprint::normalize);
+    clsFootprint.def("isNormalized", &Footprint::isNormalized);
+    clsFootprint.def("transform", &Footprint::transform,
+            "source"_a, "target"_a, "region"_a, "doClip"_a=true);
+    clsFootprint.def("findEdgePixels", &Footprint::findEdgePixels);
+    clsFootprint.def("include", &Footprint::include,
+            "children"_a, "ignore"_a=false);
+    clsFootprint.def("isPersistable", &Footprint::isPersistable);
+
+    declareMask<int>(clsFootprint);
```
Comment (fred3m) [inline python/lsst/afw/detection/footprint.cc]: Would it be better to specify what type of int (`std::int32_t`, `std::int64_t`) for this declaration and the ones below?

File: python/lsst/afw/detection/footprintCtrl.cc [hunk 2]
```diff
@@ -22,24 +22,69 @@
...
 #include <pybind11/pybind11.h>
 //#include <pybind11/operators.h>
-//#include <pybind11/stl.h>
+#include <pybind11/stl.h>
+
+#include "lsst/afw/detection/FootprintCtrl.h"
 
 namespace py = pybind11;
 
 using namespace lsst::afw::detection;
```
Comment (fred3m) [inline python/lsst/afw/detection/footprintCtrl.cc]: I think this can be removed

File: python/lsst/afw/detection/footprintMerge.cc [hunk 3]
```diff
@@ -22,24 +22,38 @@
...
+#include "lsst/afw/detection/FootprintMerge.h"
 
 namespace py = pybind11;
+using namespace py::literals;
+
+namespace lsst {
+namespace afw {
+namespace detection {
 
 using namespace lsst::afw::detection;
```
Comment (fred3m) [inline python/lsst/afw/detection/footprintMerge.cc]: please remove

File: python/lsst/afw/detection/footprintSet.cc [hunk 4]
```diff
@@ -1,45 +1,121 @@
...
 
-    /* Member types and enums */
+    /* Members */
+    declareMakeHeavy<int>(cls);
+    declareMakeHeavy<float>(cls);
+    declareMakeHeavy<double>(cls);
+    declareMakeHeavy<std::uint16_t>(cls);
+//    declareMakeHeavy<std::uint64_t>(cls);
+}
+}
```
Comment (fred3m) [inline python/lsst/afw/detection/footprintSet.cc]: please indent the declare function or comment the end of the namespace.

File: python/lsst/afw/detection/heavyFootprint.cc [hunk 5]
```diff
@@ -1,45 +1,107 @@
...
+            Footprint const &, lsst::afw::image::MaskedImage<ImagePixelT, MaskPixelT, VariancePixelT> const &,
+            HeavyFootprintCtrl const *))makeHeavyFootprint<ImagePixelT, MaskPixelT, VariancePixelT>,
+        "foot"_a, "img"_a, "ctrl"_a = nullptr);
 
-    /* Member types and enums */
+    // In Swig this seems to be suffixed with a type (i.e. mergeHeavyFootprintsF)
+    // but there really doesn't seem any good reason why that is done, so removed it
+    mod.def("mergeHeavyFootprints", mergeHeavyFootprints<ImagePixelT, MaskPixelT, VariancePixelT>);
+}
+}
```
Comment (fred3m) [inline python/lsst/afw/detection/heavyFootprint.cc]: please indent or comment the end of the namespace

File: python/lsst/afw/detection/footprint.cc (hunk 1 above)
Comment (pschella) [inline python/lsst/afw/detection/footprint.cc]: In general I'm all for specific integer types but in this case it has to match the instantiations in the `.cc` files.

File: python/lsst/afw/detection/footprintCtrl.cc (hunk 2 above)
Comment (pschella) [inline python/lsst/afw/detection/footprintCtrl.cc]: Yes

File: python/lsst/afw/detection/footprintMerge.cc (hunk 3 above)
Comment (pschella) [inline python/lsst/afw/detection/footprintMerge.cc]: Yes

File: python/lsst/afw/detection/footprintSet.cc (hunk 4 above)
Comment (pschella) [inline python/lsst/afw/detection/footprintSet.cc]: Ok

File: python/lsst/afw/detection/heavyFootprint.cc (hunk 5 above)
Comment (pschella) [inline python/lsst/afw/detection/heavyFootprint.cc]: Ok

**Pull request merged**